wordpress-mobile/WordPress-iOS
# Reader: Follow conversation should handle `success: false`

Ref. #15642 

In this PR
- the taskSucceeded bool continues to bubble up the success blocks until it reaches one where user feedback is handled
- if the user fails to follow the conversation, a toast message appears with a generic error message and reverts the Follow conversation button

**Not implemented:** better error messaging to the user. Also not implemented is a nudge for users to check their Reader subscription notification settings.  

### To test:
1. Log in with a non-a8c WordPress account, with a validated email address. 
2. On the web, disable email notifications by visiting [Reader subscription notification settings](https://wordpress.com/me/notifications/subscriptions) and checking the box in the "Block emails" section.
2. Go to Reader > Discover and find a blog post that has the comment button enabled
3. Tap the comment button, then tap the "Follow conversation by email" button
4. A toast message will appear, "Unable to subscribe to the comments" and the follow conversation button will revert from "**Following** conversation by email" to "**Follow** conversation..."

(A user will always be allowed to unfollow a conversation.)

PR submission checklist:
